roles/provision_virtual_esxi/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Check General Mandatory Variables Are Defined
+  ansible.builtin.assert:
+    that:
+      - provision_virtual_esxi_username is defined
+      - provision_virtual_esxi_password is defined
+      - provision_virtual_esxi_hostname is defined
+      - provision_virtual_esxi_vm_names is defined
+    quiet: true
+    fail_msg: Variable must be set when using this role.
+
+- name: Check If Hosts Exist
```

Comment:
Check If Hosts Exist -> Fail If Hosts Exist

Reply:
This task doesnt fail if the hosts exist, it just causes the next few tasks that deploy the hosts to be skipped. There is still an additional task at the end of the role that always powers on the hosts.
I dont think this role should fail if the hosts exist, but I can see an argument for ending the role if the hosts exist (which essentially just means not powering on the hosts at the end)